feat/add #69 Cart API 추가

kimpompu opened the pull request:
## 📌 개요
- Cart API 추가

## 🔨 변경 사항
- 단위테스트
- 카트 생성 / softDelete / 전체 삭제 API 추가

## ✅ 테스트
- [x] 단위/통합 테스트 통과 확인
- [x] 로컬 실행 후 정상 동작 확인

## ☑️ 체크리스트
- [x] 코드 컨벤션 준수
- [x] 문서화 (ERD, API 명세서 등) 반영

## 🙋 리뷰 요청사항(선택)
- (예시)이런 A라는 문제가 있는데, B와 C중에 어떤게 괜찮을까요?

kimpompu committed:
feat/add #69 Cart API 추가

- 단위테스트
- 카트 생성 / softDelete / 전체 삭제

kimpompu committed:
feat/fix Cart API

- 카트 전체조회 API 수정

skfkgla commented on src/main/java/com/sparta/delivery/backend/cart/service/CartService.java:
```diff
@@ -11,5 +30,86 @@
+		Cart cart = cartRepository.findById(cartId).orElseThrow(()->new IllegalArgumentException("존재하지 않는 메뉴이거나 이미 삭제되었습니다."));
+
+		// 로그인한 유저와 카트의 유저 아이디가 다를 경우
+		if (!cart.getCustomer().getUser().getId().equals(user.getId())) {
```
아마 조회 쿼리가 여기서 n+1 문제 때문에 2번 더 날아갈 것 같은데 fetchJoin이나 해당 조회쿼리 하나만 날리는건 어떨까요?

kimpompu replied:
네, 말씀해주신 부분 확인해보고 수정해보겠습니다.

skfkgla commented on src/test/java/com/sparta/delivery/backend/store/repository/StoreRepositoryTest.java:
```diff
@@ -0,0 +1,190 @@
+		storeA = Store.builder().name("도미노피자").reviewRate(4.8).deliveryFee(3000).minOrderPrice(15000).status(StoreStatusEnum.CLOSED).owner(owner).build();
+		em.persist(storeA);
+
+		Thread.sleep(100);
```
Thread.sleep을 통한 테스트는 좋지 않은 것 같습니다. 계속해서 쓰레드 sleep 테스트가 누적되게 되면, 실행 시간이 불필요하게 늘어날 수 있어요ㅠㅠ

명시적으로 createdAt을 할당하거나(ReflectionTestUtils 사용), 
각 저장 후 em.flush()를 호출하는 방법을 사용해보시는건 어떠신가요?
더 좋은 방법이 있다면 다른 방식으로 대체하셔도 좋습니다!

kimpompu replied:
말씀해주신 부분은 생각 못했습니다! 수정하도록 하겠습니다👍

skfkgla commented on src/test/java/com/sparta/delivery/backend/store/repository/StoreRepositoryTest.java:
```diff
@@ -0,0 +1,190 @@
+	@Autowired
+	private EntityManager em;
+
+	private User testUser;
```
해당 변수명은 Manager임을 나타내면 좋을 것 같습니다!
혹시 현재 테스트코드에 testUser를 사용중인 부분이 있을까요?

kimpompu replied:
아니요 변수명은 임의로 지정한 것이라 변경해놓도록 하겠습니다! 감사합니다😊

kimpompu committed:
feat/fix CartAPI 쿼리 수정

- CartRepository JPQL Query 추가
- StoreRepository Test Thread.sleep 제거
- CartRepository Test JPQL Query 테스트
